src/common/progressBar.js

```diff
@@ -0,0 +1,73 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { View } from 'react-native';
+import Bar from 'react-native-progress/Bar';
+import type { ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';
+import { getLoading } from '../selectors';
+import { BRAND_COLOR, NAVBAR_SIZE } from '../styles';
+import { connect } from '../react-redux';
+import type { Dispatch } from '../types';
+
+type SelectorProps = $ReadOnly<{|
+  loading: boolean,
+|}>;
+
+type Props = $ReadOnly<{|
```

Comment:
Can you remove the props you aren't currently using here? That seems to be all of them except for `dispatch` and `SelectorProps`.

And then move the parts that you're just passing into `Bar` into the JSX part of the code.

Reply:
not all of them , I defined them as props because maybe later we need progressbar or some change like we need smaller loading width than full screen, or to change color/background or loading-bar with more height(thick) so we can adjust bar with minimal props, even then if you think that needs to be removed then some can be eleminated like animated, indeterminateAnimationDuration,useNativeDriver and animationConfig I will remove these

Reply:
If we need them later, we can add them later.

For a case like this, where this component should basically always look the same across the different places it's called, it's better not to provide a lot of props like this, since if we change them somewhere, we're probably going to want to change them everywhere.